### DISCUSSION TITLE
Custom overrides of switches like `-s` on a per-slashcommand basis

### DISCUSSION THREAD

**Discussion by VanWeapon** [arnoudkooi/SN-Utils#329] [Ideas]
This thought came from the fact that `-s` flag doesn't work with `/log` command
The `-s` flag is used for opening records for the given table only for those in the current scope. 

In this case, this command doesn't work because `syslog` doesnt have a `sys_scope` field, instead it uses a Source field. 
Alternatively there is `syslog_app_scope` which does have `sys_scope`

There's probably a few ways to handle this, but I guess my question is if it's possible to override flags for specific slashcommands.

E.g. for all slashcommands, default behaviour of `-s` is to apply a `sys_scope=$scopeid` filter, however if the slashcommand is `/log` I want to change the behaviour of `-s` only for this command (to be `source=$scopename` or override distination to `syslog_app_scope` with standard filter appending)

Maybe this could be an extension of the custom switches feature referenced here: https://www.arnoudkooi.com/switch/ 

```json
{
    "s": {
        "description": "Current Scope",
        "value": "^source=javascript:gs.getCurrentScopeName()",
        "type": "encodedquerypart",
        "overwrites": ["/log"], // new part here to denote special behaviour for certain commands?
    },
}
```

**Comment by arnoudkooi** [comment]
I'll have a look, I like the idea, although I have the impression only a small subset uses the switch feature in the first place.
I would really want to restructure the slash command code to be more modular, in order to keep it maintainable.

**Comment by arnoudkooi** [comment]
I added a way to do it, let me know if this works, where the object key should be `switch.tablename`, in this case `s.syslog`
Available in 6.3.2.0. Will update when Chrome version is published.

https://github.com/arnoudkooi/ServiceNow-Utils/blob/0b415090756f73446856718f848c90df9de904d8/inject.js#L360-L362

**Reply by VanWeapon** [reply]
FYI There is a shortcut to get the scope name which is `gs.getCurrentScopeName()`
Will test when chrome version is out

**Comment by arnoudkooi** [comment]
I already checked, but that one always returns `rhino.global`
The Chrome update is now available

**Comment by VanWeapon** [comment]
Checked and this is working for me, cheers

**Comment by VanWeapon** [comment]
Just remembered this and came up with another examples where this could be useful 

`-n` on slashcommands for things like `br` or `cs` to add `sysparm_query=collection=$table` so that the table from current record is prefilled on a new script record (the name of the field is different on different tables, such as `collection` on br, `table` on client script, or `name` for dictionary and choices)

Are there any plans to make these editable via the settings page?
